### PR TITLE
[FIXED] Do not hold jetstream lock cleaning up orphans, could deadlock.

### PR DESCRIPTION
**Pull request by derekcollison**
Signed-off-by: Derek Collison <derek@nats.io>

### CODE REVIEW

File: server/jetstream_cluster.go [hunk 1]
```diff
@@ -960,36 +960,58 @@ type recoveryUpdates struct {
...
+		accName, mset, consumer := o.acc.Name, o.mset, o.name
+		o.mu.RUnlock()
+		stream := "N/A"
+		if mset != nil {
+			mset.mu.RLock()
+			stream = mset.cfg.Name
+			mset.mu.RUnlock()
+		}
+		s.Warnf("Detected orphaned consumer '%s > %s > %s', will cleanup", accName, stream, consumer)
+		o.delete()
```
Comment (Jarema) [inline server/jetstream_cluster.go:1013]: Are we sure we're ok with ignoring errors from `mset.delete()` and `o.delete()`?
Even if we are - would it make sense to log that error at least?

Comment (derekcollison) [reply server/jetstream_cluster.go:1013]: Ww could log if non-nil. Feel free to coordinate with @neilalexander and update PR.. I will be out of pocket for next few hours.

Comment (derekcollison) [reply server/jetstream_cluster.go]: ok I added them. Fixed.

**Review by neilalexander — APPROVED** [review]

**Review by Jarema — APPROVED** [review]
LGTM

**Pull request merged**